hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/TestSstFilteringService.java

```diff
@@ -0,0 +1,230 @@
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.rocksdb.LiveFileMetaData;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import static org.apache.hadoop.hdds.HddsConfigKeys.HDDS_CONTAINER_REPORT_INTERVAL;
+import static org.apache.hadoop.hdds.HddsConfigKeys.HDDS_DB_PROFILE;
+import static org.apache.hadoop.ozone.OzoneConsts.OM_DB_NAME;
+import static org.apache.hadoop.ozone.OzoneConsts.OM_KEY_PREFIX;
+import static org.apache.hadoop.ozone.OzoneConsts.OM_SNAPSHOT_DIR;
+import static org.apache.hadoop.ozone.om.OMConfigKeys.OZONE_SNAPSHOT_SST_FILTERING_SERVICE_INTERVAL;
+
+/**
+ * Test SST Filtering Service.
+ */
+public class TestSstFilteringService {
+  @Rule
+  public TemporaryFolder folder = new TemporaryFolder();
+  private OzoneManagerProtocol writeClient;
+  private OzoneManager om;
+  private static final Logger LOG =
+      LoggerFactory.getLogger(TestSstFilteringService.class);
+
+  @BeforeClass
+  public static void setup() {
+    ExitUtils.disableSystemExit();
+  }
+
+  private OzoneConfiguration createConfAndInitValues() throws IOException {
+    OzoneConfiguration conf = new OzoneConfiguration();
+    File newFolder = folder.newFolder();
+    if (!newFolder.exists()) {
+      Assert.assertTrue(newFolder.mkdirs());
+    }
+    System.setProperty(DBConfigFromFile.CONFIG_DIR, "/");
+    ServerUtils.setOzoneMetaDirPath(conf, newFolder.toString());
+    conf.setTimeDuration(HDDS_CONTAINER_REPORT_INTERVAL, 200,
+        TimeUnit.MILLISECONDS);
+    conf.setTimeDuration(OZONE_SNAPSHOT_SST_FILTERING_SERVICE_INTERVAL, 100,
+        TimeUnit.MILLISECONDS);
+    conf.setEnum(HDDS_DB_PROFILE, DBProfile.TEST);
+    conf.setQuietMode(false);
+
+    return conf;
+  }
+
+  @After
+  public void cleanup() throws Exception {
+    om.stop();
+  }
+
+  /**
+   * Test checks whether for existing snapshots
+   * the checkpoint should not have any sst files that do not correspond to
+   * the bucket on which create snapshot command was issued.
+   *
+   * The SSTFiltering service deletes only the last level of
+   * sst file (rocksdb behaviour).
+   *
+   * 1. Create Keys for vol1/buck1 (L0 ssts will be created for vol1/buck1)
+   * 2. compact the db (new level SSTS will be created for vol1/buck1)
+   * 3. Create keys for vol1/buck2 (L0 ssts will be created for vol1/buck2)
+   * 4. Take snapshot on vol1/buck2.
+   * 5. The snapshot will contain compacted sst files pertaining to vol1/buck1
+   *    Wait till the BG service deletes these.
+   *
+   * @throws IOException - on Failure.
+   */
+
+  @Test
+  public void testIrrelevantSstFileDeletion()
```

Comment:
why didn't this test catch the bug where we were deleting valid keys ? Can we add more validation tests around this service here  ?

Reply:
I only checked for the non-existence of the irrelevant files , will add an assertion to check existence of the relevant files.

Reply:
Done.